### PR TITLE
fix: Fix address checkbox behaviour

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it fixes the issue where an address was being saved as the default shipping/billing address, even when the checkbox was not checked.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/app/[locale]/(checkout)/checkout/(checkout-details)/shipping-address/_forms/create-form.tsx [hunk 1]
```diff
@@ -23,12 +24,14 @@ import { createCheckoutShippingAddress } from "./actions";
...
 export const CreateShippingAddressForm = ({
+  addressFormRows,
+  addresses,
   checkout,
   countries,
-  addressFormRows,
   countryCode,
 }: {
   addressFormRows: readonly AddressFormRow[];
+  addresses: FormattedAddress[];
```
Comment (piotrgrundas) [inline apps/storefront/src/app/[locale]/(checkout)/checkout/(checkout-details)/shipping-address/_forms/create-form.tsx:34]: Maybe just `saveForFutre: boolean` since you don't do anything with it?

**Review by piotrgrundas — APPROVED** [review]

**Review by piotrgrundas — APPROVED** [review]
